core/gui/src/app/workspace/component/menu/menu.component.ts

```diff
@@ -253,17 +253,27 @@ export class MenuComponent implements OnInit {
    * get the html to export all results.
    */
   public onClickGenerateReport(): void {
-    //Get notification
+    // Get notification
     this.notificationService.info("The report is being generated...");
 
     const workflowName = this.currentWorkflowName;
+    const payload: WorkflowContent = this.workflowActionService.getWorkflowContent();
+
+    // Extract operatorIDs and operators from the parsed payload
+    const operatorIds = payload.operators.map((operator: { operatorID: string }) => operator.operatorID);
+    const operators = payload.operators;
 
     // Invokes the method of the report printing service
     this.reportGenerationService
       .generateWorkflowSnapshot(workflowName)
       .pipe(untilDestroyed(this))
       .subscribe({
-        next: (snapshot: string) => this.reportGenerationService.generateReportAsHtml(snapshot, workflowName),
+        next: (workflowSnapshotURL: string) =>
+          this.reportGenerationService.getAllOperatorResults({
```

Comment:
here, you first call functions to get the operator result. And then you pass the result to generateReportHtml function.